eureka/src/main/java/com/linecorp/armeria/server/eureka/EurekaUpdatingListener.java

```diff
@@ -187,10 +189,10 @@ private void scheduleHeartBeat(EventLoop eventLoop, InstanceInfo newInfo) {
                                              TimeUnit.SECONDS);
     }
 
-    private InstanceInfo fillAndCreateNewInfo(InstanceInfo oldInfo, Server server) {
+    private static InstanceInfo fillAndCreateNewInfo(InstanceInfo oldInfo, Server server) {
         final String defaultHostname = server.defaultHostname();
         final String hostName = oldInfo.getHostName() != null ? oldInfo.getHostName() : defaultHostname;
-        appName = oldInfo.getAppName() != null ? oldInfo.getAppName() : hostName;
```

Comment:
this field mutation was indescribable in the function name, so I moved it out and made the function static to prevent a recurrence!